pkg/auth/auth.go

```diff
@@ -0,0 +1,81 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/fastly/cli/pkg/api"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/text"
+)
+
+// ignoreCommands represents commands that do not require a token.
+var ignoreCommands = []string{"configure", "ip-list", "update", "version"}
+
+// Required determines if the OAuth flow is required.
+func Required(cmd, token, apiEndpoint string, s config.Source, stdout io.Writer, cf api.ClientFactory) (required bool) {
+	for _, name := range ignoreCommands {
+		if cmd == name {
+			return false
+		}
+	}
+
+	// No token is defined so yes we require the OAuth flow.
+	if s == config.SourceUndefined {
+		text.Info(stdout, "An access token is required but has not been provided.")
+		return true
+	}
+
+	if s == config.SourceFile {
+		c, err := cf(token, apiEndpoint)
+		if err != nil {
+			return true
+		}
+		if ok := ValidateToken(c, stdout); !ok {
+			required = true
+		}
+	}
+
+	// False (no authentication flow required).
+	return required
+}
```

Comment:
Here is where we determine if an authentication flow is required.

- If the command doesn't use a token, then _**no**_ an auth flow is not required.
- If no token exists in the CLI config, then _**yes**_ an auth flow is required.
- If a token exists in the CLI config, but is invalid or expired, then _**yes**_ an auth flow is required.

This means, if a user provides a token via `--token` then we skip the authentication flow.  
This allows for greater flexibility, and as a side-effect gives us an easy way to skip auth in our test suite.